src/input-field.tsx

```diff
@@ -122,7 +124,7 @@ export const InputField = React.forwardRef<HTMLInputElement | HTMLTextAreaElemen
                 ref={ref}
                 checked={checked}
                 onChange={onChange}
-                disabled={readOnly}
+                disabled={propsDisabled}
```

Comment:
we need to leave readOnly setting the field to be disabled.  You can mark an entire form as readOnly and it'll set all it's fields the same.   What we can do though is make it something like:
`disabled={Boolean(readonly || propsDisabled)}`